ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -549,7 +549,18 @@ PrjFS_Result PrjFS_DeleteFile(
     CombinePaths(s_virtualizationRootFullPath.c_str(), relativePath, fullPath);
     
     struct stat path_stat;
-    stat(fullPath, &path_stat);
+    if (0 != stat(fullPath, &path_stat))
+    {
+        switch(errno)
+        {
+            case ENOENT:  // A component of fullPath does not exist
+            case ENOTDIR: // A component of fullPath is not a directory
```

Comment:
Do you know when it would hit this case? How does that stat know that you are looking for a directory?  What if this was called for a file?  I don't think in VFSForGit we ever do try to delete a file though.

Reply:
> Do you know when it would hit this case?

I would expect the `ENOTDIR` case to be hit if you called `stat` on a path like /a/b/c/d.txt and a/b/c is a file.

> How does that stat know that you are looking for a directory?
> What if this was called for a file?

`stat` doesn't know what you're looking for, these errors indicate that the file that you're looking for is not on disk (either because part of the path does not exist, or part of the path is a file).

> I don't think in VFSForGit we ever do try to delete a file though.

In VFS4G we delete files when changing to a projection where some of the file placeholders no longer exist.